### DISCUSSION TITLE
hjkl vs arrows

### DISCUSSION THREAD

**Discussion by ThatStoney** [mokshadharma/vim-beyond-wq#4] [Q&A] [answered]
I really enjoyed your DoL presentation @mokshadharma - thank you! You've inspired me to level up my `vim` game, and I have a question about the use of `hjkl` vs arrow keys. For touch typists `hjkl` makes a lot of sense because that's the home position and the most used keys, `jk`, are under dominant fingers.  I am a fairly fast typist, but I'm not a touch typist and have always used the arrow keys in `vim`. Is there any advantage for non-touch typists like me to start using `hjkl` instead? It won't be easy for me to relearn after all these years, but I'm willing to do so if there are advantages I'm unaware of.

**Comment by mokshadharma** [comment, marked as answer]
The main advantages are:

1. A lot less arm movement to the arrow keys and back to the home row.
2. The rest of the keys you use in vim and to type are back on the main keyboard, not by the arrow keys, so it makes for less movement of the arm in general if you want to type or use the rest of vim if you keep your fingers mostly on the home row (for non-touch-typists, "keeping your fingers on the home row" refers to keeping your left-hand fingers on ASDF and your right on JKL; on the QWERTY keyboard layout, which happens to be the layout for which vi and vim were designed).

So it all boils down to efficiency of vim use and efficiency of typing.  You'll do both a lot faster if you keep your fingers mostly on the home row.

That said, I do use the arrow keys myself in vim, but it's very very rare, and I have their functions remapped to do something else that I very rarely need, so using them does not impact my efficiency very much.  However, using them for something as common as ordinary movement will both slow you down a lot and also cause a lot of unnecessary wear on your arm.

**Comment by ThatStoney** [comment]
Thanks @mokshadharma, I appreciate the detailed reply. I've worked my way through `vimtutor` a couple times now and forced myself to not touch the arrow keys. I'm slowly getting the feel for it, but I'll need more practice before it's more natural.

**Reply by mokshadharma** [reply]
It's great that you are so motivated.  Stick to it and it'll feel natural with practice.
